lib/maths/CTimeSeriesDecompositionDetail.cc

```diff
@@ -481,7 +481,7 @@ CTimeSeriesDecompositionDetail::CPeriodicityTest::CPeriodicityTest(const CPeriod
     // Note that m_Windows is an array.
     for (std::size_t i = 0u; !isForForecast && i < other.m_Windows.size(); ++i) {
         if (other.m_Windows[i]) {
-            m_Windows[i] = std::make_shared<CExpandingWindow>(*other.m_Windows[i]);
+            m_Windows[i].reset(new CExpandingWindow{*other.m_Windows[i]});
```

Comment:
I know `make_unique` is C++14, but can we maybe use a backport, which we can throw away when we switched?

Herb Sutter has an implementation here: https://herbsutter.com/gotw/_102/

Reply:
Another possibility for using `std::make_unique` is to change the `-std` argument for the Mac/Linux compilers to `c++14`/`gnu++14` respectively on the 6.x branch.

Even though Visual Studio 2013 doesn't support very much of C++14, one bit it _does_ support is `std::make_unique`.  So switching the standard filtering used by the other compilers would mean all of them support `std::make_unique`.

People would just need to remember that only a tiny fraction of C++14 is available in the 6.x branch.  From https://msdn.microsoft.com/en-us/library/hh567368.aspx:

> Visual C++ in Visual Studio 2013 implemented some key C++14 library features:
> * "Transparent operator functors" `less<>`, `greater<>`, `plus<>`, `multiplies<>`, and so on.
> * `make_unique<T>(args...)` and `make_unique<T[]>(n)`
> * `cbegin()`/`cend()`, `rbegin()`/`rend()`, and `crbegin()`/`crend()` non-member functions.

Reply:
I wondered if `boost::make_unique` might return `boost::unique_ptr`s, but just checked and it really does create `std::unique_ptr`s.  So I guess it was designed for exactly this case of a compiler supporting `std::unique_ptr` but not `std::make_unique`.

I'm happy to use it until 6.x is for bug fixes only and then someone can do a search and replace of `boost::make_unique` with `std::make_unique` on the master branch.

Reply:
Ok cool, I'll leave as I have it.